src/window/settings.rs

```diff
@@ -49,10 +60,36 @@ impl Default for WindowSettings {
     }
 }
 
+#[cfg(target_os = "macos")]
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct OptionAsMeta(pub OptionAsAlt);
+
+#[cfg(target_os = "macos")]
+impl ParseFromValue for OptionAsMeta {
+    fn parse_from_value(&mut self, value: Value) {
+        let s = value.as_str().unwrap();
+        match OptionAsAlt::deserialize(s.into_deserializer()) as Result<OptionAsAlt, value::Error> {
+            Ok(oa) => *self = OptionAsMeta(oa),
+            Err(e) => error!("Setting neovide_input_macos_option_key_is_meta expected one of OnlyLeft, OnlyRight, Both, or None, but received {:?}: {:?}", e, value),
+        };
+    }
+}
+
+#[cfg(target_os = "macos")]
+impl From<OptionAsMeta> for Value {
+    fn from(oam: OptionAsMeta) -> Self {
+        let s = serde_json::to_string(&oam.0).unwrap();
```

Comment:
i feel a little uncertain about this (i.e. whether it's appropriate to use the JSON serializer here).

more broadly, i don't really know if i took the right approach to serialization/deserialization.

Reply:
...but maybe this is secondary to the topic of whether we want to try a different approach to settings updates as mentioned in the comment in `WinitWindowWrapper::new()`?

Reply:
I don't think this is needed, we don't really need to call `set`, just read the values and use them. `ParseFromValue` is still needed though.